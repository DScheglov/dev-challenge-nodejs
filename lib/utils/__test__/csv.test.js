const toCsv = require('../csv');

describe('toCsv', () => {
  it('should return a single row', () => {
    expect(toCsv([[1, 2, 3]])).toMatchSnapshot();
  });

  it('should return a multiple rows', () => {
    expect(toCsv([[1, 2, 3], [3, 2, 1]])).toMatchSnapshot();
  });

  it('should support headers', () => {
    expect(toCsv([['Field 1', 'Field 2', 'Field 3'], [1, 2, 3]])).toMatchSnapshot();
  });

  it('should return a multiple rows and header row', () => {
    expect(toCsv([['Field #1', 'Field #2', 'Field #3'], [1, 2, 3], [3, 2, 1]])).toMatchSnapshot();
  });

  it('should process a complex data', () => {
    const data = [
      ['Value With "Quotes"', 'Value With ,(comma)', 1, 2, 3],
      ['Just a Value', 'Value with \n carret return', 1, 2, 3],
    ];

    expect(toCsv(data)).toMatchSnapshot();
  });
});
