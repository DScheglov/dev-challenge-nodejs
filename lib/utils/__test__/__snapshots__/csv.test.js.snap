// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toCsv should process a complex data 1`] = `
"\\"Value With \\"\\"Quotes\\"\\"\\",\\"Value With ,(comma)\\",1,2,3
Just a Value,\\"Value with 
 carret return\\",1,2,3"
`;

exports[`toCsv should return a multiple rows 1`] = `
"1,2,3
3,2,1"
`;

exports[`toCsv should return a multiple rows and header row 1`] = `
"Field #1,Field #2,Field #3
1,2,3
3,2,1"
`;

exports[`toCsv should return a single row 1`] = `"1,2,3"`;

exports[`toCsv should support headers 1`] = `
"Field 1,Field 2,Field 3
1,2,3"
`;
